`--overwrite` broken for `ctapipe-train-energy-regressor`

**Describe the bug**
```
ctapipe-train-energy-regressor \
	-i data/gamma-diffuse_train_en.dl2.h5 --output=build-0.17.1.dev593+g5fa96da8/energy.pkl \
	-c train_energy_regressor.yml \
	--provenance-log=build-0.17.1.dev593+g5fa96da8/energy.pkl.provlog \
	--log-file=build/train_energy.log \
	--log-level=INFO \
	--overwrite
2023-02-02 14:17:20,586 INFO [ctapipe.ctapipe-train-energy-regressor] (tool.initialize): Loading config from '[PosixPath('/home/mnoethe/Uni/CTA/test_ml/train_energy_regressor.yml')]'
2023-02-02 14:17:20,628 INFO [ctapipe.ctapipe-train-energy-regressor] (tool.initialize): ctapipe version 0.17.1.dev593+g5fa96da8
2023-02-02 14:17:20,628 INFO [ctapipe.ctapipe-train-energy-regressor] (tool.run): Starting: ctapipe-train-energy-regressor
2023-02-02 14:17:20,711 ERROR [ctapipe.ctapipe-train-energy-regressor] (tool.run): Config option `overwrite` not recognized by `CrossValidator`.
2023-02-02 14:17:20,712 ERROR [ctapipe.ctapipe-train-energy-regressor] (tool.run): Use --help for more info
make: *** [Makefile:19: build-0.17.1.dev593+g5fa96da8/energy.pkl] Error 2
```

**To Reproduce**

Run `ctapipe-train-energy-regressor` with `--overwrite`

**Expected behavior**

Works

--- a/ctapipe/conftest.py
+++ b/ctapipe/conftest.py
@@ -525,6 +525,7 @@ def energy_regressor_path(model_tmp_path):
                 f"--output={out_file}",
                 f"--config={config}",
                 "--log-level=INFO",
+                "--overwrite",
             ],
         )
         assert ret == 0
@@ -586,6 +587,7 @@ def particle_classifier_path(model_tmp_path, gamma_train_clf, proton_train_clf):
                 f"--output={out_file}",
                 f"--config={config}",
                 "--log-level=INFO",
+                "--overwrite",
             ],
         )
         assert ret == 0

--- a/ctapipe/tools/train_disp_reconstructor.py
+++ b/ctapipe/tools/train_disp_reconstructor.py
@@ -60,16 +60,6 @@ class TrainDispReconstructor(Tool):
         ),
     ).tag(config=True)
 
-    flags = {
-        "overwrite": (
-            {
-                "TrainDispReconstructor": {"overwrite": True},
-                "CrossValidator": {"overwrite": True},
-            },
-            "Overwrite existing output",
-        ),
-    }
-
     aliases = {
         ("i", "input"): "TableLoader.input_url",
         ("o", "output"): "TrainDispReconstructor.output_path",

--- a/ctapipe/tools/train_energy_regressor.py
+++ b/ctapipe/tools/train_energy_regressor.py
@@ -56,16 +56,6 @@ class TrainEnergyRegressor(Tool):
         default_value=0, help="Random seed for sampling and cross validation"
     ).tag(config=True)
 
-    flags = {
-        "overwrite": (
-            {
-                "TrainEnergyReconstructor": {"overwrite": True},
-                "CrossValidator": {"overwrite": True},
-            },
-            "Overwrite existing output",
-        ),
-    }
-
     aliases = {
         ("i", "input"): "TableLoader.input_url",
         ("o", "output"): "TrainEnergyRegressor.output_path",

--- a/ctapipe/tools/train_particle_classifier.py
+++ b/ctapipe/tools/train_particle_classifier.py
@@ -82,16 +82,6 @@ class TrainParticleClassifier(Tool):
         help="Random number seed for sampling and the cross validation splitting",
     ).tag(config=True)
 
-    flags = {
-        "overwrite": (
-            {
-                "TrainParticleClassifier": {"overwrite": True},
-                "CrossValidator": {"overwrite": True},
-            },
-            "Overwrite existing output",
-        ),
-    }
-
     aliases = {
         "signal": "TrainParticleClassifier.input_url_signal",
         "background": "TrainParticleClassifier.input_url_background",

